pyspectral/rayleigh.py

```diff
@@ -237,7 +239,20 @@ def _do_interp(minterp, sunzsec, azidiff, satzsec):
         res = minterp(interp_points2)
         return res.reshape(sunzsec.shape)
 
-    def get_reflectance(self, sun_zenith, sat_zenith, azidiff, bandname, redband=None):
+    @staticmethod
+    def _do_reduce_effect(zenith, rayref, thresh_zen, maxzen, strength):
+        """Reduce the Rayleigh correction amount at high zenith angles.
+        This linearly scales the Rayleigh reflectance, `rayref`, for solar or satellite zenith angles, `zenith`,
+        above a threshold angle, `thresh_zen`. Between `thresh_zen` and `maxzen` the Rayleigh reflectance will
+        be linearly scaled, from one at `thresh_zen` to zero at `maxzen`.
+        """
+        factor = 1. - strength * where(zenith < thresh_zen, 0, (zenith - thresh_zen) / (maxzen - thresh_zen))
+        # For low zenith factor can be greater than one, so we need to clip it into a sensible range.
+        factor = clip(factor, 0, 1)
+        return rayref * factor
+
+    def get_reflectance(self, sun_zenith, sat_zenith, azidiff, bandname, redband=None,
+                        reduce_hizen=None, reduce_strength=1.):
```

Comment:
Could we combine `reduce_hizen` and `reduce_strength` into one parameter? eg if we kept only reduce_strength, setting it to 0 could be equivalent to skipping this?

Also some explanation in the docstring on what the new argument(s) mean would be nice.